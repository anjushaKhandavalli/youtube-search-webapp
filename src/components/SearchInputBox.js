import React, {useState} from 'react';
import { SearchBarContainer, SearchBar, Field, Label, Input  } from "./SearchInputBox.style";
import { SEARCH_CONSTANTS } from "../constants";

const SearchInputBox = ( {handleSubmit}) => {
  const [searchText, setSearchText] = useState('');

  const onSubmit = event => {
    event.preventDefault();
    handleSubmit(searchText);
  }
  return(
    <SearchBarContainer>
      <SearchBar onSubmit={onSubmit}>
        <Field>
          <Label 
            htmlFor="search-box" 
            data-testid="search-box-label"
            >
              {SEARCH_CONSTANTS.LABEL}
          </Label>
          <br/>
          <Input
            data-testid="search-box-input"
            name="search-box"
            type="text"
            placeholder={SEARCH_CONSTANTS.PLACEHOLDER}
            value={searchText}
            onChange={(event) => setSearchText(event.target.value)}
          />
        </Field>
      </SearchBar>
    </SearchBarContainer>
  )
}

export default SearchInputBox;