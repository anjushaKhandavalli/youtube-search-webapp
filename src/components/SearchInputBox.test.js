import React from "react";
import { render } from "@testing-library/react";
import SearchInputBox from "./SearchInputBox";

describe('SearchInputBox', () => {
  it('should render the search input box with label and input box',() => {
    const {container, findByTestId} = render(<SearchInputBox/>);

    expect(container).toMatchSnapshot();
    expect(findByTestId('search-box-label')).toBeDefined();
    expect(findByTestId('search-box-input')).toBeDefined();
  })
});