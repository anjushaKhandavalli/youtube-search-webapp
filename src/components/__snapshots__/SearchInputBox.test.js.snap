// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInputBox should render the search input box with label and input box 1`] = `
<div>
  <div
    class="sc-bdfBwQ hSckpy"
  >
    <form
      class="sc-gsTCUz fOtSmi"
    >
      <div>
        <label
          class="sc-dlfnbm koWLPh"
          data-testid="search-box-label"
          for="search-box"
        >
          Search for videos via keywords:
        </label>
        <br />
        <div
          class="sc-jSgupP iaInnZ"
        >
          <input
            class="sc-hKgILt kFahnm"
            data-testid="search-box-input"
            name="search-box"
            placeholder="Ex: dogs,cars etc.."
            type="text"
            value=""
          />
          <button
            class="sc-eCssSg dBXRyi"
          >
            Search
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
